internal/command/action.go

```diff
@@ -176,12 +195,52 @@ EnvLoop:
 
 	setupEnvFile(ac.Args, env, &tempFactory)
 
+	// Setup Docker args
+	var argsWithDockerArgs []string
+	for _, arg := range ac.Args {
+		if arg == DOCKER_ARGS_MAGIC {
+			// Replace argument with slice of docker options
+			argsWithDockerArgs = append(argsWithDockerArgs, dockerArgs...)
+			continue
+		}
+
+		// TODO: we need to decide which of these if we want to support (2)
+		// 1. summon [...] docker run @SUMMONDOCKERARGS [...], replace only entire top-level arg. The
+		// top-level arg is replaced by is replaced by N>0 args equating to @SUMMONDOCKERARGS.
+		// 2. summon ... sh -c "docker run @SUMMONDOCKERARGS [...]", also replace substrings
+		// inside args but the replacement is as a single string.
+		//
+		// The code below should support (2). There'll be some ambiguity though...
```

Comment:
I think we should support replacing substrings. I think lots of users wrap their subcommands with `bin/bash -ec '...'`,  and it would be better for us to continue to support that syntax. I personally can't think of a use case where someone would benefit from using that syntax over Summon directly calling docker, but it may be easier to ensure all docker subcommands are processed correctly.

Reply:
I agree, we should include this support. Maybe there's some weird docker-in-docker use case where we're running summon inside a container that requires a `bash/sh -c ...`???

Reply:
Uncomment, added feature and test case.